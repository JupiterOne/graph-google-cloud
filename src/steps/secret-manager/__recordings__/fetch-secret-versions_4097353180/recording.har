{
  "log": {
    "_recordingName": "fetch-secret-versions",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "6.0.5"
    },
    "entries": [
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 717,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/x-www-form-urlencoded"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "717"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 300,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/x-www-form-urlencoded",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 482,
          "content": {
            "encoding": "utf-8",
            "mimeType": "application/json; charset=UTF-8",
            "size": 482,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Fri, 19 Aug 2022 14:31:32 GMT"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 506,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-08-19T14:31:32.439Z",
        "time": 84,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 84
        }
      },
      {
        "_id": "a5c156251fd2a9cdee36bc6130e24899",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/5.0.2 gl-node/14.20.0 auth/7.1.1"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/5.0.2 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "secretmanager.googleapis.com"
            }
          ],
          "headersSize": 1359,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://secretmanager.googleapis.com/v1/projects/j1-gc-integration-dev-v3/secrets"
        },
        "response": {
          "bodySize": 378,
          "content": {
            "encoding": "utf-8",
            "mimeType": "application/json; charset=UTF-8",
            "size": 378,
            "text": "{\"secrets\":[{\"name\":\"projects/167984947943/secrets/topsecret\",\"replication\":{\"automatic\":{}},\"createTime\":\"2022-08-10T13:39:10.097472Z\",\"etag\":\"\\\"15e5e32c7a3840\\\"\"},{\"name\":\"projects/167984947943/secrets/topsecret2\",\"replication\":{\"automatic\":{}},\"createTime\":\"2022-08-10T14:20:01.704458Z\",\"etag\":\"\\\"15e5e3be9ac60a\\\"\"}],\"totalSize\":2}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Fri, 19 Aug 2022 14:31:33 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 483,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-08-19T14:31:32.531Z",
        "time": 396,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 396
        }
      },
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 1,
        "cache": {},
        "request": {
          "bodySize": 717,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/x-www-form-urlencoded"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "717"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 300,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/x-www-form-urlencoded",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 414,
          "content": {
            "encoding": "utf-8",
            "mimeType": "application/json; charset=UTF-8",
            "size": 414,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Fri, 19 Aug 2022 14:31:33 GMT"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 506,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-08-19T14:31:32.936Z",
        "time": 96,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 96
        }
      },
      {
        "_id": "ee68a9c6d094f7cccbfc1d99f1f9bec8",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/5.0.2 gl-node/14.20.0 auth/7.1.1"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/5.0.2 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "secretmanager.googleapis.com"
            }
          ],
          "headersSize": 1366,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://secretmanager.googleapis.com/v1/projects/167984947943/secrets/topsecret/versions"
        },
        "response": {
          "bodySize": 427,
          "content": {
            "encoding": "utf-8",
            "mimeType": "application/json; charset=UTF-8",
            "size": 427,
            "text": "{\"versions\":[{\"name\":\"projects/167984947943/secrets/topsecret/versions/2\",\"createTime\":\"2022-08-10T14:19:20.840979Z\",\"state\":\"ENABLED\",\"replicationStatus\":{\"automatic\":{}},\"etag\":\"\\\"15e5e3bc2b3f13\\\"\",\"clientSpecifiedPayloadChecksum\":true},{\"name\":\"projects/167984947943/secrets/topsecret/versions/1\",\"createTime\":\"2022-08-10T13:39:12.521669Z\",\"state\":\"ENABLED\",\"replicationStatus\":{\"automatic\":{}},\"etag\":\"\\\"15e5e32c9f35c5\\\"\",\"clientSpecifiedPayloadChecksum\":true}],\"totalSize\":2}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Fri, 19 Aug 2022 14:31:33 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 483,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-08-19T14:31:33.037Z",
        "time": 449,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 449
        }
      },
      {
        "_id": "9a0e93e5a844b367873f540b744fa1de",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/5.0.2 gl-node/14.20.0 auth/7.1.1"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/5.0.2 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "secretmanager.googleapis.com"
            }
          ],
          "headersSize": 1367,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://secretmanager.googleapis.com/v1/projects/167984947943/secrets/topsecret2/versions"
        },
        "response": {
          "bodySize": 414,
          "content": {
            "encoding": "utf-8",
            "mimeType": "application/json; charset=UTF-8",
            "size": 414,
            "text": "{\"versions\":[{\"name\":\"projects/167984947943/secrets/topsecret2/versions/1\",\"createTime\":\"2022-08-10T14:20:04.160121Z\",\"state\":\"ENABLED\",\"replicationStatus\":{\"automatic\":{}},\"etag\":\"\\\"15e5e3bec03e79\\\"\",\"clientSpecifiedPayloadChecksum\":true}],\"totalSize\":1}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Fri, 19 Aug 2022 14:31:34 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 483,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-08-19T14:31:33.494Z",
        "time": 439,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 439
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
