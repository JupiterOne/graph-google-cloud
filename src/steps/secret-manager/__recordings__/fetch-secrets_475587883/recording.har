{
  "log": {
    "_recordingName": "fetch-secrets",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "6.0.5"
    },
    "entries": [
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 717,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/x-www-form-urlencoded"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "717"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 300,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/x-www-form-urlencoded",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 428,
          "content": {
            "encoding": "utf-8",
            "mimeType": "application/json; charset=UTF-8",
            "size": 428,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Fri, 19 Aug 2022 14:31:32 GMT"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 506,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-08-19T14:31:31.790Z",
        "time": 131,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 131
        }
      },
      {
        "_id": "a5c156251fd2a9cdee36bc6130e24899",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/5.0.2 gl-node/14.20.0 auth/7.1.1"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/5.0.2 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "secretmanager.googleapis.com"
            }
          ],
          "headersSize": 1359,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://secretmanager.googleapis.com/v1/projects/j1-gc-integration-dev-v3/secrets"
        },
        "response": {
          "bodySize": 320,
          "content": {
            "encoding": "utf-8",
            "mimeType": "application/json; charset=UTF-8",
            "size": 320,
            "text": "{\"secrets\":[{\"name\":\"projects/167984947943/secrets/topsecret\",\"replication\":{\"automatic\":{}},\"createTime\":\"2022-08-10T13:39:10.097472Z\",\"etag\":\"\\\"15e5e32c7a3840\\\"\"},{\"name\":\"projects/167984947943/secrets/topsecret2\",\"replication\":{\"automatic\":{}},\"createTime\":\"2022-08-10T14:20:01.704458Z\",\"etag\":\"\\\"15e5e3be9ac60a\\\"\"}],\"totalSize\":2}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Fri, 19 Aug 2022 14:31:32 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 483,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-08-19T14:31:31.934Z",
        "time": 445,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 445
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
