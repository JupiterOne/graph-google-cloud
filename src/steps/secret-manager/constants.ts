import { RelationshipClass } from '@jupiterone/integration-sdk-core';

export const SecretManagerSteps = {
  FETCH_SECRETS: {
    id: 'fetch-secrets',
    name: 'Fetch Secret Manager secrets',
  },
  FETCH_SECRET_VERSIONS: {
    id: 'fetch-secret-versions',
    name: 'Fetch Secret Manager secret versions',
  },
};

export const SecretManagerEntities = {
  SECRET: {
    resourceName: 'Secret',
    _type: 'google_secret_manager_secret',
    _class: ['Group'],
  },
  SECRET_VERSION: {
    resourceName: 'Secret Version',
    _type: 'google_secret_manager_secret_version',
    _class: ['Secret'],
  },
};

export const SecretManagerRelationships = {
  SECRET_HAS_VERSION: {
    _type: 'google_secret_manager_secret_has_version',
    sourceType: SecretManagerEntities.SECRET._type,
    _class: RelationshipClass.HAS,
    targetType: SecretManagerEntities.SECRET_VERSION._type,
  },
};
