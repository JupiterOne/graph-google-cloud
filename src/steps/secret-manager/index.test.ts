import {
  createMockStepExecutionContext,
  executeStepWithDependencies,
  StepTestConfig,
} from '@jupiterone/integration-sdk-testing';
import { fetchSecrets, fetchSecretVersions } from '.';
import { IntegrationConfig, invocationConfig } from '../..';
import { integrationConfig } from '../../../test/config';

import { Recording, setupGoogleCloudRecording } from '../../../test/recording';
import { SecretManagerEntities, SecretManagerSteps } from './constants';

describe('#secret-manager', () => {
  let recording: Recording;

  afterEach(async () => {
    if (recording) await recording.stop();
  });

  test('fetch-secrets', async () => {
    recording = setupGoogleCloudRecording({
      name: 'fetch-secrets',
      directory: __dirname,
    });

    const stepTestConfig: StepTestConfig = {
      stepId: SecretManagerSteps.FETCH_SECRETS.id,
      instanceConfig: integrationConfig,
      invocationConfig: invocationConfig as any,
    };

    const result = await executeStepWithDependencies(stepTestConfig);
    expect(result).toMatchStepMetadata(stepTestConfig);
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchSecrets(context);
    await fetchSecretVersions(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(
      context.jobState.collectedEntities.filter(
        (e) => e._type === SecretManagerEntities.SECRET._type,
      ),
    ).toMatchGraphObjectSchema({
      _class: ['Group'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'google_secret_manager_secret' },
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          createdAt: { type: 'number' },
          expiresAt: { type: 'number' },
          automaticReplicationKmsKeyName: { type: 'number' },
          nextRotationTime: { type: 'number' },
          rotationPeriod: { type: 'string' },
          topicNames: { type: 'array', items: { type: 'string' } },
          ttl: { type: 'string' },
        },
      },
    });

    expect(
      context.jobState.collectedEntities.filter(
        (e) => e._type === SecretManagerEntities.SECRET_VERSION._type,
      ),
    ).toMatchGraphObjectSchema({
      _class: ['Secret'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'google_secret_manager_secret_version' },
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          createdAt: { type: 'number' },
          destroyTime: { type: 'number' },
        },
      },
    });
  });
});
