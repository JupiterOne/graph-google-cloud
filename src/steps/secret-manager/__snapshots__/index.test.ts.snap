// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#secret-manager should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Group",
      ],
      "_key": "projects/167984947943/secrets/topsecret",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "createTime": "2022-08-10T13:39:10.097472Z",
            "name": "projects/167984947943/secrets/topsecret",
            "replication": Object {
              "automatic": Object {},
            },
          },
        },
      ],
      "_type": "google_secret_manager_secret",
      "automaticReplicationKmsKeyName": undefined,
      "createdAt": 1660138750097,
      "createdOn": undefined,
      "displayName": "projects/167984947943/secrets/topsecret",
      "expiresAt": undefined,
      "name": "projects/167984947943/secrets/topsecret",
      "nextRotationTime": undefined,
      "rotationPeriod": undefined,
      "topicNames": undefined,
      "ttl": undefined,
    },
    Object {
      "_class": Array [
        "Group",
      ],
      "_key": "projects/167984947943/secrets/topsecret2",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "createTime": "2022-08-10T14:20:01.704458Z",
            "name": "projects/167984947943/secrets/topsecret2",
            "replication": Object {
              "automatic": Object {},
            },
          },
        },
      ],
      "_type": "google_secret_manager_secret",
      "automaticReplicationKmsKeyName": undefined,
      "createdAt": 1660141201704,
      "createdOn": undefined,
      "displayName": "projects/167984947943/secrets/topsecret2",
      "expiresAt": undefined,
      "name": "projects/167984947943/secrets/topsecret2",
      "nextRotationTime": undefined,
      "rotationPeriod": undefined,
      "topicNames": undefined,
      "ttl": undefined,
    },
    Object {
      "_class": Array [
        "Secret",
      ],
      "_key": "projects/167984947943/secrets/topsecret/versions/2",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "createTime": "2022-08-10T14:19:20.840979Z",
            "name": "projects/167984947943/secrets/topsecret/versions/2",
            "state": "ENABLED",
          },
        },
      ],
      "_type": "google_secret_manager_secret_version",
      "createdAt": 1660141160840,
      "createdOn": undefined,
      "destroyTime": undefined,
      "displayName": "projects/167984947943/secrets/topsecret/versions/2",
      "name": "projects/167984947943/secrets/topsecret/versions/2",
    },
    Object {
      "_class": Array [
        "Secret",
      ],
      "_key": "projects/167984947943/secrets/topsecret/versions/1",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "createTime": "2022-08-10T13:39:12.521669Z",
            "name": "projects/167984947943/secrets/topsecret/versions/1",
            "state": "ENABLED",
          },
        },
      ],
      "_type": "google_secret_manager_secret_version",
      "createdAt": 1660138752521,
      "createdOn": undefined,
      "destroyTime": undefined,
      "displayName": "projects/167984947943/secrets/topsecret/versions/1",
      "name": "projects/167984947943/secrets/topsecret/versions/1",
    },
    Object {
      "_class": Array [
        "Secret",
      ],
      "_key": "projects/167984947943/secrets/topsecret2/versions/1",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "createTime": "2022-08-10T14:20:04.160121Z",
            "name": "projects/167984947943/secrets/topsecret2/versions/1",
            "state": "ENABLED",
          },
        },
      ],
      "_type": "google_secret_manager_secret_version",
      "createdAt": 1660141204160,
      "createdOn": undefined,
      "destroyTime": undefined,
      "displayName": "projects/167984947943/secrets/topsecret2/versions/1",
      "name": "projects/167984947943/secrets/topsecret2/versions/1",
    },
  ],
  "collectedRelationships": Array [
    Object {
      "_class": "HAS",
      "_fromEntityKey": "projects/167984947943/secrets/topsecret",
      "_key": "projects/167984947943/secrets/topsecret|has|projects/167984947943/secrets/topsecret/versions/2",
      "_toEntityKey": "projects/167984947943/secrets/topsecret/versions/2",
      "_type": "google_secret_manager_secret_has_version",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "projects/167984947943/secrets/topsecret",
      "_key": "projects/167984947943/secrets/topsecret|has|projects/167984947943/secrets/topsecret/versions/1",
      "_toEntityKey": "projects/167984947943/secrets/topsecret/versions/1",
      "_type": "google_secret_manager_secret_has_version",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "projects/167984947943/secrets/topsecret2",
      "_key": "projects/167984947943/secrets/topsecret2|has|projects/167984947943/secrets/topsecret2/versions/1",
      "_toEntityKey": "projects/167984947943/secrets/topsecret2/versions/1",
      "_type": "google_secret_manager_secret_has_version",
      "displayName": "HAS",
    },
  ],
  "encounteredTypes": Array [
    "google_secret_manager_secret",
    "google_secret_manager_secret_version",
    "google_secret_manager_secret_has_version",
  ],
  "numCollectedEntities": 5,
  "numCollectedRelationships": 3,
}
`;
